src/rider/main/kotlin/com/jetbrains/rider/plugins/efcore/features/migrations/add/OpenMigrationFileService.kt

```diff
@@ -2,48 +2,55 @@ package com.jetbrains.rider.plugins.efcore.features.migrations.add
 
 import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
 import com.intellij.openapi.fileEditor.FileEditorManager
-import com.intellij.openapi.fileEditor.OpenFileDescriptor
 import com.intellij.openapi.project.Project
-import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.rd.util.lifetime
+import com.intellij.openapi.vfs.VfsUtil
 import com.intellij.openapi.vfs.VirtualFile
-import com.jetbrains.rider.plugins.efcore.features.shared.flushFileSystem
-import com.jetbrains.rider.plugins.efcore.rd.MigrationIdentity
-import com.jetbrains.rider.plugins.efcore.rd.riderEfCoreModel
-import com.jetbrains.rider.projectView.solution
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
-import kotlin.io.path.Path
-import kotlin.io.path.pathString
+import com.intellij.openapi.vfs.isFile
+import com.jetbrains.rd.util.lifetime.LifetimeDefinition
+import com.jetbrains.rd.util.reactive.adviseOnce
+import com.jetbrains.rd.util.reactive.filter
 
 @Service(Service.Level.PROJECT)
-class OpenMigrationFileService(private val intellijProject: Project) {
-    suspend fun tryOpenMigrationFile(migrationFolderPath: String, migrationIdentity: MigrationIdentity): Boolean {
-        intellijProject.flushFileSystem(migrationFolderPath)
+class OpenMigrationFileService(val intellijProject: Project) {
 
-        val beModel = intellijProject.solution.riderEfCoreModel
-        val migration = withContext(Dispatchers.Main) {
-            beModel.getMigration.startSuspending(migrationIdentity)
-        }
+    private companion object {
+        private const val MIGRATION_FILE_NAME_PATTERN = "^\\d{14}_%s.cs$"
+    }
 
-        if (migration == null)
-            return false
+    var fileListenerLifetimeDef: LifetimeDefinition? = null
+    var expectedFileNamePattern: String? = null
+    var migrationsFolderPath: String? = null
```

Comment:
Do we really need these 2 to keep for later use? Looks like we could just make them local variables in `startOpeningFile`